1-letter username causing `panic: strings: negative Repeat count`

### What happened?

I have a one-letter username for my AdGuard Home login. When adding it to `origin` in `adguardhome-sync.yaml`:
```yaml
origin:
  url: http://192.168.10.123
  username: a
  password: mypassword
```
The container fails to boot due to an error: `panic: strings: negative Repeat count`.

The bug seems to be on [this line](https://github.com/bakito/adguardhome-sync/blob/main/pkg/types/types.go#L193) where `len(s)-2` resolves to `-1` because my one-letter username has a length of `1`.

### AdguardHome-Sync Version

`v0.7.1`, Docker

### AdguardHome Version

`v0.107.57`, Docker

### OS Information

Ubuntu 22.04.5 LTS, x86_64

### Configuration

```yaml
origin:
  url: http://192.168.10.123
  username: a
  password: mypassword

replicas:
  - url: http://192.168.10.44
    username: b
    password: mypassword2
```

### Current Applied Configuration

```shell
—
```

### Relevant log output

```shell
2025-03-13T09:54:33.157Z        INFO    sync    sync/sync.go:38 AdGuardHome sync        {"version": "v0.7.1", "build": "2025-02-25T20:27:09Z", "os": "linux", "arch": "amd64"}
panic: strings: negative Repeat count

goroutine 1 [running]:
strings.Repeat({0xec7040?, 0xa0?}, 0xd39660?)
        /usr/local/go/src/strings/strings.go:624 +0x585
github.com/bakito/adguardhome-sync/pkg/types.mask({0xeed4a0, 0x1})
        /go/src/app/pkg/types/types.go:193 +0x50
github.com/bakito/adguardhome-sync/pkg/types.(*AdGuardInstance).Mask(0xc000158f20)
        /go/src/app/pkg/types/types.go:165 +0x25
github.com/bakito/adguardhome-sync/pkg/types.(*Config).mask(0xc000158c60?)
        /go/src/app/pkg/types/types.go:112 +0x4f
github.com/bakito/adguardhome-sync/pkg/types.(*Config).Log(0x0?, 0xc0000a4460)
        /go/src/app/pkg/types/types.go:106 +0x25
github.com/bakito/adguardhome-sync/pkg/sync.Sync(0xc000158c60)
        /go/src/app/pkg/sync/sync.go:39 +0x1cd
github.com/bakito/adguardhome-sync/cmd.init.func1(0x14a6460, {0xd68bb5?, 0x4?, 0xd68bb9?})
        /go/src/app/cmd/run.go:38 +0x27b
github.com/spf13/cobra.(*Command).execute(0x14a6460, {0xc0003239a0, 0x2, 0x2})
        /go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1015 +0xa94
github.com/spf13/cobra.(*Command).ExecuteC(0x14a61a0)
        /go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x40c
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071
github.com/bakito/adguardhome-sync/cmd.Execute()
        /go/src/app/cmd/root.go:27 +0x1a
main.main()
        /go/src/app/main.go:6 +0xf
```

### Anything else?

_No response_

--- a/Makefile
+++ b/Makefile
@@ -20,6 +20,9 @@ fmt: tb.golines tb.gofumpt
 # Run tests
 test: generate fmt lint test-ci
 
+fuzz:
+	 go test -fuzz=FuzzMask -v ./pkg/types/ -fuzztime=60s
+
 # Run ci tests
 test-ci: mocks tidy tb.ginkgo
 	$(TB_GINKGO) --cover --coverprofile coverage.out.tmp ./...

--- a/pkg/types/types.go
+++ b/pkg/types/types.go
@@ -187,8 +187,8 @@ func (i *AdGuardInstance) Init() error {
 }
 
 func mask(s string) string {
-	if s == "" {
-		return "***"
+	if len(s) < 3 {
+		return strings.Repeat("*", len(s))
 	}
 	mask := strings.Repeat("*", len(s)-2)
 	return fmt.Sprintf("%v%s%v", string(s[0]), mask, string(s[len(s)-1]))

--- a/pkg/types/types_fuzz_test.go
+++ b/pkg/types/types_fuzz_test.go
@@ -1,0 +1,15 @@
+package types
+
+import (
+	"testing"
+)
+
+func FuzzMask(f *testing.F) {
+	testcases := []string{"", "a", "ab", "abc", "abcd"}
+	for _, tc := range testcases {
+		f.Add(tc)
+	}
+	f.Fuzz(func(t *testing.T, value string) {
+		_ = mask(value)
+	})
+}

--- a/pkg/types/types_test.go
+++ b/pkg/types/types_test.go
@@ -83,6 +83,15 @@ var _ = Describe("Types", func() {
 				Ω(masked.API.Password).Should(Equal("p**s"))
 			})
 		})
+		DescribeTable("mask should work correctly",
+			func(value, expected string) {
+				Ω(mask(value)).Should(Equal(expected))
+			},
+			Entry(`Empty password`, "", ""),
+			Entry(`1 char password`, "a", "*"),
+			Entry(`2 char password`, "ab", "**"),
+			Entry(`3 char password`, "abc", "a*c"),
+		)
 	})
 	Context("Feature", func() {
 		Context("LogDisabled", func() {

